ios/engine/KMEI/KeymanEngine/Classes/Resource Management/ResourceDownloadManager.swift

```diff
@@ -321,43 +324,137 @@ public class ResourceDownloadManager {
+    let now = date.timeIntervalSince1970
+    let timeDelta = now - timestamp
+
+    return timeDelta <= ResourceDownloadManager.DISTRIBUTION_CACHE_VALIDITY_THRESHOLD
+  }
+
+  /**
+   * Indicates whether any packages are known to be updatable based on the most recently
+   * cached package-version query for each.  To bypass the cache and query for updated results,
+   * use `queryKeysForUpdatablePackages` instead.
+   */
   public var updatesAvailable: Bool {
     get {
-      return getAvailableUpdates() != nil
+      return getKeysForUpdatablePackages().count > 0
+    }
+  }
+
+  /**
+   * Returns a Set of package keys for packages known to be updatable,
+   * based upon the results of their most recently-run latest-version queries.
+   *
+   * If neither `KeymanPackage.queryCurrentVersions` nor `KeymanPackage.queryDistributionStates`
+   * has been run for a package, it is assumed to be current, as this is a synchronous method.
+   */
+  public func getKeysForUpdatablePackages() -> Set<KeymanPackage.Key> {
+    let packages = ResourceFileManager.shared.installedPackages
+
+    let keys: [KeymanPackage.Key] = packages.compactMap { package in
+      guard package.versionQueryCache.distributionMethod == .cloud,
+            let versionString = package.versionQueryCache.latestVersion,
+            let latestVersion = Version(versionString) else {
+        // We only update packages with full cloud support.
+        return nil
+      }
+
+      if package.version < latestVersion {
+        return package.key
+      } else {
+        return nil
+      }
+    }
+
+    return Set(keys)
+  }
+
+  /**
+   * Queries the cloud for the latest version of every installed package, returning a Set
+   * of keys for packages with available updates.  The results will also be cached for use
+   * with related `get`-prefixed methods, like `getKeysForUpdatablePackages`.
+   */
+  public func queryKeysForUpdatablePackages(completionBlock: @escaping (Set<KeymanPackage.Key>?, Error?) -> Void) {
+    let allPackages = ResourceFileManager.shared.installedPackages
+
+    // This call will automatically cache the results for future use / access by the 'get' equivalent.
+    KeymanPackage.queryCurrentVersions(for: allPackages.map { $0.key }, withSession: session) { results, error in
+      guard error == nil, let results = results else {
+        completionBlock(nil, error)
+        return
+      }
+
+      let updatables: [KeymanPackage.Key] = allPackages.compactMap { package in
+        guard let result = results[package.key] else {
+          return nil
+        }
+
+        return package.version < result ? package.key : nil
+      }
+
+      completionBlock(Set(updatables), nil)
     }
   }
 
+  // Possibly worth not deprecating, as its continued existence doesn't exactly hurt anything.
+  // But its intended partner method will no longer exist, so... yeah.
+  @available(*, deprecated, message: "Deprecated in favor of `getKeysForUpdatablePackages`.")
```

Comment:
When was `getAvailableUpdates()` introduced? Is it in the public Keyman Engine API?

Reply:
First introduced in [13.0](https://github.com/keymanapp/keyman/blob/stable-13.0/ios/engine/KMEI/KeymanEngine/Classes/Resource%20Management/ResourceDownloadManager.swift#L499).

The method is public, so it was technically accessible.  It does not appear to have been documented on help.keyman.com.

Reply:
It was refactored out of the keyboard picker as part of https://github.com/keymanapp/keyman/pull/1979.

Reply:
> It does not appear to have been documented on help.keyman.com.

In that case, I reckon we delete it rather than keep it around. It was undocumented so not part of the API contract.

Reply:
Removed as part of #3372.